### PR TITLE
Implementing Sentry Tracking, Fallback Phone functionality, Alert Reason Disambiguation

### PR DESCRIPTION
**Pull request by sampathsatti**
Features have been tested on development server. Looking to deploy changes to production.

### CODE REVIEW

File: db/005-human_read_location.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    migrationId := 5;
+
+    -- Get the migration ID of the last file to be successfully run
+    SELECT MAX(id) INTO lastSuccessfulMigrationId
+    FROM migrations;
+
+    -- Only execute this script if its migration ID is next after the last successful migration ID
+    IF migrationId - lastSuccessfulMigrationId = 1 THEN
+        -- ADD SCRIPT HERE
+        ALTER TABLE locations ADD COLUMN location_human text default 'Bermuda';
```
Comment (trisapeace) [inline db/005-human_read_location.sql:15]: Way to get https://www.youtube.com/watch?v=mP07Oyr7enQ stuck in my head ;-)

File: db/db.js [hunk 2]
```diff
@@ -178,9 +178,27 @@ async function getMostRecentSession(locationid) {
   }
 }
 
+// Gets session with a specific SessionID
+
+async function getSessionWithSessionId(sessionid){
+  try{
+    const results = await pool.query("SELECT * FROM sessions WHERE sessionid = $1", [sessionid]);
```
Comment (trisapeace) [inline db/db.js:185]: Please try to use single quotes for Javascript.

File: db/db.js [hunk 3]
```diff
@@ -178,9 +178,27 @@ async function getMostRecentSession(locationid) {
...
+  }
+  catch(e){
+    console.log(`Error running the getSessionWithSessionId query: ${e}`);
+  }
+}
+
 
 // Gets the last session data in the table for a specified phone number
-async function getMostRecentSessionPhone(phone) {
+async function getMostRecentSessionPhone(phone){
```
Comment (trisapeace) [inline db/db.js:201]: Funny that you changed this. I actually prefer it with the space.

File: db/db.js [hunk 4]
```diff
@@ -365,6 +383,17 @@ async function isOverdoseSuspected(xethru, session, location) {
     let condition2 = 1 * (session.still_counter > still_threshold); //seconds
     let condition3 = 1 * (sessionDuration > sessionDuration_threshold);
     let condition4 = 1 * (0);
+
+    let alertReason='NotSet'
```
Comment (trisapeace) [inline db/db.js:387]: It would be more consistent with our formatting to have spaces around the `=`

File: db/db.js [hunk 5]
```diff
@@ -365,6 +383,17 @@ async function isOverdoseSuspected(xethru, session, location) {
...
+
+    if(condition2 == 1){
+      alertReason = 'Duration';
+    }
+
+    if( (condition1==1) || (condition3==1) ){
```
Comment (trisapeace) [inline db/db.js:393]: Same here. It would be more consistent with our formatting to have spaces around the `==`s

File: db/redis.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
+require('dotenv').config();
 const Redis = require("ioredis");
 const radarData = require('./radarData.js');
 const doorData = require('./doorData.js');
 const stateData = require('./stateData.js');
-require('dotenv').config();
 const client = new Redis(6379, process.env.REDIS_CLUSTER_IP); // uses defaults unless given configuration object
```
Comment (trisapeace) [inline db/redis.js:6]: Is this value the same for production and test?
Do you not need to call the `getEnvVar` function here?

File: index.js [hunk 7]
```diff
@@ -472,63 +463,82 @@ io.on('connection', (socket) => {
...
+  }
+  //else do nothing
 }
 
-//Heartbeat Helper Functions
+async function fallbackMessage(sessionid) {
+  console.log("Fallback message being sent");
+  let session = await db.getSessionWithSessionId(sessionid); // Gets the updated state for the chatbot
+  if(session.chatbot_state == STATE.WAITING_FOR_RESPONSE) {
+    console.log("Fallback if block");
```
Comment (trisapeace) [inline index.js:511]: Did you mean to leave these logs in?

**Review by trisapeace — APPROVED** [review]
Looks great!
My comments are just details, nothing that actually needs to change for this to be good to go.

File: index.js [hunk 8]
```diff
@@ -472,63 +463,82 @@ io.on('connection', (socket) => {
...
+    let locationData = await db.getLocationData(session.locationid)
+    console.log(`fallback number is:  ${locationData.fallback_phonenumber}`)
+    console.log(`twilio number is:  ${locationData.twilio}`)
+    await sendTwilioMessage(locationData.twilio_number, locationData.fallback_number,`An alert to check on the washroom at ${locationData.location_human} was not responded to. Please check on it`)
```
Comment (trisapeace) [inline index.js:515]: When was `fallback_number` added to the DB? I don't see it in any of the migration scripts.

File: db/redis.js (hunk 6 above)
Comment (sampathsatti) [reply db/redis.js:6]: Are you referring to `REDIS_CLUSTER_IP`? That's different in production and test, we use different env files when making the docker container for the respective deployments.

File: db/db.js (hunk 5 above)
Comment (sampathsatti) [reply db/db.js:393]: Noting this, and will add a linter to the list of things in the next PR.

**Comment by sampathsatti** [comment]
Merging, Theresa's comments will be integrated into the code in the next pull request - They are not expected to functionally change code execution.

File: db/redis.js (hunk 6 above)
Comment (trisapeace) [reply db/redis.js:6]: Yes, that was the one that I was thinking of. And I think I see the difference now. It looks like it only matter when you have automated tests running the code and you don't have any tests yet that use this file. So maybe it doesn't matter.

File: db/db.js (hunk 5 above)
Comment (trisapeace) [reply db/db.js:393]: Cool. This is the ESLint configuration that we're currently using in Buttons and the shared library:
https://github.com/bravetechnologycoop/brave-alert-lib/blob/main/.eslintrc.json

However, I know that @rhoberman was looking into sets of common ESLint config that we can use across all our projects. I'm not sure what ended up happening with that investigation. It looks like BeSafe is currently using this: https://github.com/bravetechnologycoop/besafe-app/blob/master/.eslintrc.json which is definitely different.

To make it run in Travis you can:
1. Add an `npm run lint` script to `package.json` like this: https://github.com/bravetechnologycoop/brave-alert-lib/blob/main/package.json#L11
2. Tell Travis to run it each time like this: https://github.com/bravetechnologycoop/brave-alert-lib/blob/main/.travis.yml#L13

**Pull request merged**